sources/Map/Map.cpp

```diff
@@ -43,15 +44,34 @@ namespace BBM
 			return;
 		health->takeDmg(health->getHealthPoint());
 	}
+	
+	void MapGenerator::wallYouShouldNotPass(WAL::Entity &entity,
+	                                const WAL::Entity &wall,
+	                                CollisionComponent::CollidedAxis collidedAxis)
+	{
+		auto *mov = entity.tryGetComponent<MovableComponent>();
+
+		if (!mov)
+			return;
+		if (collidedAxis & CollisionComponent::CollidedAxis::X)
+			mov->_velocity.x = 0;
+		if (collidedAxis & CollisionComponent::CollidedAxis::Y)
+			mov->_velocity.y = 0;
+		if (collidedAxis & CollisionComponent::CollidedAxis::Z)
+			mov->_velocity.z = 0;
+	}
 
 	void MapGenerator::wallCollided(WAL::Entity &entity,
 	                                const WAL::Entity &wall,
 	                                CollisionComponent::CollidedAxis collidedAxis)
 	{
 		auto *mov = entity.tryGetComponent<MovableComponent>();
+		auto *playerBonus = entity.tryGetComponent<PlayerBonusComponent>();
 
 		if (!mov)
 			return;
+		if (playerBonus && playerBonus->isNoClipOn)
+			return;
 		if (collidedAxis & CollisionComponent::CollidedAxis::X)
```

Comment:
You should call wallYouShouldNotPass here, duplicating the code is not really good.